python/gcp_client/evaluation_tools/gcp_client/gcp.py

```diff
@@ -0,0 +1,216 @@
+"""
+================================
+Google Cloud Platform NWM Client
+================================
+This module provides classes that offer a convenient 
+interface to retrieve National Water Model (NWM) data 
+from Google Cloud Platform.
+
+https://console.cloud.google.com/marketplace/details/noaa-public/national-water-model
+
+Classes
+-------
+    NWMDataService
+
+"""
+
+from google.cloud import storage
+from io import BytesIO
+import xarray as xr
+import pandas as pd
+from os import cpu_count
+from multiprocessing import Pool
+
+def NWM_bytes_to_DataFrame(bytes_string) -> pd.DataFrame:
+    """Convert bytes from an NWM channel route netcdf4 file to a 
+    pandas.DataFrame
+
+    Parameters
+    ----------
+    bytes_string : bytes, required
+        Raw bytes from NWM channel route file.
+
+    Returns
+    -------
+    df : pandas.DataFrame
+        A stacked DataFrame.
+    
+    """
+    # Load data as xarray DataSet
+    ds = xr.load_dataset(BytesIO(bytes_string), engine='h5netcdf', 
```

Comment:
I think we can use the `drop_variables` keyword argument here to drop `['time', 'reference_time', 'crs', 'nudge', 'velocity', 'qSfcLatRunoff', 'qBucket', 'qBtmVertRunoff']`. It might be worth testing how this affects performance. 

To note, if you provide a variable that is not present in the file it is ignored, no warnings or exceptions.